cli/tauri-bundler/src/bundle/osx_bundle.rs

```diff
@@ -87,9 +89,215 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
     create_bootstrapper(&bundle_directory, settings)
       .with_context(|| "Failed to create OSX bootstrapper")?;
   }
+
+  if let Some(identity) = settings.osx_signing_identity() {
+    sign(app_bundle_path.clone(), identity, &settings)?;
+    match notarize_auth_args() {
+      Ok(args) => {
+        notarize(app_bundle_path.clone(), args, settings)?;
+      }
+      Err(e) => {
+        common::print_info(format!("skipping app notarize, {}", e.to_string()).as_str())?;
+      }
+    }
+  }
+
   Ok(vec![app_bundle_path])
 }
 
+fn sign(app_bundle_path: PathBuf, identity: &str, settings: &Settings) -> crate::Result<()> {
+  common::print_info(format!(r#"signing app with identity "{}""#, identity).as_str())?;
+  let mut args = vec!["--deep", "--force", "-s", identity, "--options", "runtime"];
```

Comment:
Are `--deep` and `--force` necessary here?

![image](https://user-images.githubusercontent.com/19519564/107857507-b3c2b400-6dfc-11eb-90eb-7d8563e52370.png)

https://developer.apple.com/library/archive/technotes/tn2206/_index.html#//apple_ref/doc/uid/DTS40007919-CH1-TNTAG404

Reply:
it's better to do it manually? :(

Reply:
I'm not entirely sure why 😄  I just know Apple really doesn't recommend it https://developer.apple.com/forums/thread/128166

Reply:
want me to change it? I'd do anything to get this PR done :D

Reply:
Yeah, I don't think those are necessary unless we find that it doesn't work without `--deep` and `--force`. The fact that an Apple employee wrote an article called [`--deep` Considered Harmful](https://developer.apple.com/forums/thread/129980) sounds really concerning 😂

Reply:
absolutely
I thought --deep was necessary to sign all files
and --force I think I was just using because I was testing signing on a previously signed app :joy: